### PR TITLE
Fix: optimism & l2 networks gas handling

### PR DESCRIPTION
**Pull request by chk0912**
# Gas on L2 networks

## Description

This PR aims to fix some weird gas situations using L2 chains. There are some informational scripts below. Here is what we fix for each case:

- **zkSync**: There was an issue with the block gas limit; consequently, we underestimated all the transaction's gas limit values.
- **Optimism**, **Arbitrum**, **Scroll**: We were using fixed gas limit values designed exclusively for Ethereum as default values. We discarded that value for the case of an L2 network.
- **Gnosis**: Not an L2, it seems to be calculating fees ok.

L2 netoworks: https://ethereum.org/en/layer-2/

Optimism gas data: https://community.optimism.io/docs/developers/build/transaction-fees/#stuff-to-keep-in-mind

Arbitrum gas data: https://developer.offchainlabs.com/faqs/how-fees

## Extra

I have set **Optimism** and **Gnosis** as pre-loaded networks.

### CODE REVIEW

**Review by leablock — APPROVED** [review]

**Review by julianariel — APPROVED** [review]
great stuff 🔥 works great and simple! added trivial comments but LGTM 🚀

File: packages/background/src/utils/constants/networks.ts [hunk 1]
```diff
@@ -215,7 +215,10 @@ export const INITIAL_NETWORKS: Networks = {
             decimals: 18,
         },
         isCustomNetwork: true,
```
[the remark this answers is not in the document]
Comment (julianariel) [reply packages/background/src/utils/constants/networks.ts:217]: are we setting it to `true` for all L2 networks, right? probably something more accurate related to that like `isLayer2Network`

[the remark this answers is not in the document]
Comment (fpezzola) [reply packages/background/src/utils/constants/networks.ts:217]: What do you mean by saying "this is not Ethereum"? 

That the network is not Ethereum Mainnet? If that is the case, I'd simply remove the flag and check that the chainId is different from 1.

[the remark this answers is not in the document]
Comment (fpezzola) [reply packages/background/src/utils/constants/networks.ts:217]: But if it is not the case. I would simply rename the property to `hasFixedGasCost`

**Review by fpezzola — APPROVED** [review]
LGTM!! This is great!!

**Pull request merged**